core/src/main/java/com/netease/arctic/table/BasicTableBuilder.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.table;
+
+import com.netease.arctic.ams.api.TableFormat;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.Transaction;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+import java.util.Map;
+
+
+/**
+ * A base class of interface {@link TableBuilder}
+ * @param <ThisT> self class
+ */
+public abstract class BasicTableBuilder<ThisT extends TableBuilder> implements TableBuilder {
```

Comment:
Should the `BasicArcticCatalog#ArcticTableBuilder` extend this basic class too?

Reply:
In the future, this class will inherit from this parent class, but this part of the change is not involved in the current PR, and there will be other PRs to do this in the future